tests/Helper/ObjectParserTest.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Validator\Tests\Helper;
+
+use PHPUnit\Framework\TestCase;
+use Yiisoft\Validator\Helper\ObjectParser;
+
+final class ObjectParserTest extends TestCase
+{
+    public function dataSkipStaticProperties(): array
+    {
+        return [
+            [
+                ['a' => 4, 'b' => 2],
+                new class () {
+                    public int $a = 4;
+                    public static int $b = 2;
+                },
+                false,
+            ],
+            [
+                ['a' => 4, 'c' => 'hello'],
+                new class () {
+                    public int $a = 4;
+                    public static int $b = 2;
+                    public string $c = 'hello';
+                },
+                true,
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider dataSkipStaticProperties
+     */
+    public function testSkipStaticProperties(array $expectedData, object $object, bool $skipStaticProperties): void
+    {
+        $parser = new ObjectParser($object, skipStaticProperties: $skipStaticProperties);
+
+        $this->assertSame($expectedData, $parser->getData());
+    }
+
+    public function testSkipStaticPropertiesDefault(): void
+    {
+        $object = new class () {
+            public int $a = 4;
+            public static int $b = 2;
+        };
+
+        $parser = new ObjectParser($object);
+
+        $this->assertSame(['a' => 4, 'b' => 2], $parser->getData());
+    }
```

Comment:
```suggestion
    public function dataGetData(): array
    {
        $object = new class () {
            public int $a = 4;
            public static int $b = 2;
            public string $c = 'hello';
        };

        return [
            [
                ['a' => 4, 'b' => 2, 'c' => 'hello'],
                $object,
                null,
            ],
            [
                ['a' => 4, 'b' => 2, 'c' => 'hello'],
                $object,
                false,
            ],
            [
                ['a' => 4, 'c' => 'hello'],
                $object,
                true,
            ],
        ];
    }

    /**
     * @dataProvider dataGetData
     */
    public function testGetData(array $expectedData, object $object, bool|null $skipStaticProperties): void
    {
        $arguments = [$object];
        if ($skipStaticProperties !== null) {
            $arguments['skipStaticProperties'] = $skipStaticProperties;
        }

        $parser = new ObjectParser(...$arguments);
        $this->assertSame($expectedData, $parser->getData());
    }
```

Reply:
Additional conditions — more difficult code. Suggest keep as is.